browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -1453,35 +1454,42 @@ void RewardsDOMHandler::OnPublisherListNormalized(
 void RewardsDOMHandler::GetTransactionHistory(
     const base::ListValue* args) {
   ads_service_->GetTransactionHistory(base::Bind(
-      &RewardsDOMHandler::OnTransactionHistory,
+      &RewardsDOMHandler::OnGetTransactionHistory,
       weak_factory_.GetWeakPtr()));
 }
 
-void RewardsDOMHandler::OnTransactionHistory(
-    double estimated_pending_rewards,
-    uint64_t next_payment_date_in_seconds,
-    uint64_t ad_notifications_received_this_month) {
-  if (web_ui()->CanCallJavascript()) {
-    base::DictionaryValue history;
-
-    history.SetDouble("adsEstimatedPendingRewards",
-        estimated_pending_rewards);
-
-    if (next_payment_date_in_seconds == 0) {
-      history.SetString("adsNextPaymentDate", "");
-    } else {
-      base::Time next_payment_date =
-          base::Time::FromDoubleT(next_payment_date_in_seconds);
-      history.SetString("adsNextPaymentDate",
-          base::TimeFormatWithPattern(next_payment_date, "MMMd"));
-    }
+void RewardsDOMHandler::OnGetTransactionHistory(
+    const bool success,
+    const double estimated_pending_rewards,
+    const uint64_t next_payment_date_in_seconds,
+    const uint64_t ad_notifications_received_this_month) {
+  if (!success) {
```

Comment:
shouldn't we return this to front end so that FE can decide if to retry or something?

Reply:
Retry logic is handled by the ads library where the UI is informed if the transaction history has changed, i.e. after initialization or when a user receives a reward